### PR TITLE
[nana] Fix fontconfig dependency

### LINKED ISSUE #15490: [fontconfig] build failure
**Host Environment**
- OS: Linux Manjaro
- Compiler: g++

**To Reproduce**
`./vcpkg install fontconfig`

**Failure logs** 
```--- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Generating configure for x64-linux
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:108 (message):
    Command failed: /usr/bin/autoreconf -vfi
    Working Directory: /home/venny/projects/vctest/vcpkg/buildtrees/fontconfig/src/75c13b2421-4651fd65ce.clean/
    Error code: 1
    See logs for more information:
      /home/venny/projects/vctest/vcpkg/buildtrees/fontconfig/autoconf-x64-linux-out.log
      /home/venny/projects/vctest/vcpkg/buildtrees/fontconfig/autoconf-x64-linux-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:536 (vcpkg_execute_required_process)
  ports/fontconfig/portfile.cmake:25 (vcpkg_configure_make)
  scripts/ports.cmake:136 (include)


Error: Building package fontconfig:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: fontconfig:x64-linux
  Vcpkg version: 2020.11.12-unknownhash
```

The mentioned log files are attached.
[autoconf-x64-linux-err.log](https://github.com/microsoft/vcpkg/files/5779601/autoconf-x64-linux-err.log)
[autoconf-x64-linux-out.log](https://github.com/microsoft/vcpkg/files/5779602/autoconf-x64-linux-out.log)

**Additional context**
I tried installing the nana package, but it failed at fontconfig.
Installing fontconfig directly renders the same result.

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- What does your PR fix? Fixes #15490

Update `fontconfig `dependency as the follows:

```
 find_package(Fontconfig)
 target_link_libraries(nana PUBLIC Fontconfig::Fontconfig)
```

Note: No feature needs to test.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
@Neumann-A 

Could you please help review this PR?

Thanks.

File: ports/nana/config.cmake.in [hunk 1]
```diff
@@ -4,7 +4,7 @@ if(UNIX)
   find_package(Threads REQUIRED)
   find_package(Freetype REQUIRED)
   find_package(X11 REQUIRED)
```
Comment (Neumann-A) [inline ports/nana/config.cmake.in:6]: these should also be find_dependency and make sure you have included the macro file for find_dependency

**Review by JackBoosY — APPROVED** [review]

File: ports/nana/config.cmake.in [hunk 2]
```diff
@@ -1,27 +1,28 @@
 @PACKAGE_INIT@
+include(CMakeFindDependencyMacro)
 
 if(UNIX)
-  find_package(Threads REQUIRED)
-  find_package(Freetype REQUIRED)
-  find_package(X11 REQUIRED)
-  find_library(FONTCONFIG_LIB NAMES fontconfig)
+  find_dependency(Threads REQUIRED)
```
Comment (Neumann-A) [inline ports/nana/config.cmake.in:5]: you can remove all REQUIRED keywords. find_dependency will figure it out for you

Comment (JackBoosY) [reply ports/nana/config.cmake.in:5]: find_dependency doesn't need keyword REQUIRED.

Comment (NancyLi1013) [reply ports/nana/config.cmake.in]: Thanks again. Have updated.

**Pull request merged**